lib_xud/src/core/included/XUD_Token_In_DI.S

```diff
@@ -92,17 +109,16 @@ DoneTail:
     bf          r11, XUD_IN_DoneTx
 
 SetupReceiveHandShake:
-    ldc        r11, 8
-    setpsc     res[RXD], r11                       // Set port shift count (expect 8 bit handshake)
+    {setsr      0x1                                 // enable events so we can get a handshake before waiteu
```

Comment:
Dont we need to do the setpsc before the setsr?

Reply:
why?

Reply:
by default, the event will fire if RxD receives 4 bytes. If, say, 2 bytes are received then then event wont fire. The later setpsc will reset things, meaning another 1 byte is needed to trigger the event. With an early setpsc the data and event is captured "early" firing when the setsr is run.